cli/src/ui/messages/extension/utils.ts

```diff
@@ -21,10 +21,39 @@ export function parseToolData(message: ExtensionChatMessage): ToolData | null {
 }
 
 /**
- * Parse MCP server data from message
+ * Type guard to check if an object is valid McpServerData
+ */
+export function isMcpServerData(obj: any): obj is McpServerData {
```

Comment:
We have both `ajv` and `zod` in our codebase, we should probably use one of those (`zod` if you ask me) instead of writing a custom fn.

(separate discussion: why do we have both and should one be removed :smile:)